src/utils/uint256.cairo

```diff
@@ -0,0 +1,25 @@
+from starkware.cairo.common.uint256 import Uint256, uint256_eq, uint256_le, uint256_sub, uint256_mul
+from starkware.cairo.common.bool import FALSE
+
+// @notice Internal exponentiation of two 256-bit integers.
+// @dev The result is modulo 2^256.
+// @param a The base.
+// @param b The exponent.
+// @return The result of the exponentiation.
+func uint256_exp{range_check_ptr}(a: Uint256, b: Uint256) -> Uint256 {
```

Comment:
note: we need to implement fast powering, or this will never work for high values of b

Reply:
we need indeed, I've just moved the implementation from the opcode to a util